addon_imps/tests/test_mendeley.py

```diff
@@ -0,0 +1,129 @@
+        self.assertEqual(result, "12345")
+        self.mendeley_imp.network.GET.assert_called_with("profiles/me")
+
+    def test_list_root_collections(self):
+        mock_response_data = [
+            {"id": "1", "name": "Collection 1"},
+            {"id": "2", "name": "Collection 2"},
+        ]
+        self.mendeley_imp.network.GET.return_value.__aenter__.return_value.json_content = AsyncMock(
+            return_value=mock_response_data
+        )
+
+        result = asyncio.run(self.mendeley_imp.list_root_collections())
+
+        expected_result = ItemSampleResult(
+            items=[
+                ItemResult(
+                    item_id="1", item_name="Collection 1", item_type=ItemType.COLLECTION
+                ),
+                ItemResult(
+                    item_id="2", item_name="Collection 2", item_type=ItemType.COLLECTION
+                ),
+            ],
+            total_count=2,
+        )
+
+        self.assertEqual(result, expected_result)
+        self.mendeley_imp.network.GET.assert_called_with("folders")
+
+    def test_list_collection_items(self):
+        mock_document_ids = [{"id": "doc1"}, {"id": "doc2"}]
+        mock_doc1_details = {
+            "id": "doc1",
+            "title": "Doc Title 1",
+            "type": "document",
+            "path": [],
+        }
+        mock_doc2_details = {
+            "id": "doc2",
+            "title": "Doc Title 2",
+            "type": "document",
+            "path": [],
+        }
+
+        self.mendeley_imp.network.GET.side_effect = [
+            AsyncMock(
+                __aenter__=AsyncMock(
+                    return_value=AsyncMock(
+                        json_content=AsyncMock(return_value=mock_document_ids)
+                    )
+                )
+            ),
+            AsyncMock(
+                __aenter__=AsyncMock(
+                    return_value=AsyncMock(
+                        json_content=AsyncMock(return_value=mock_doc1_details)
+                    )
+                )
+            ),
+            AsyncMock(
+                __aenter__=AsyncMock(
+                    return_value=AsyncMock(
+                        json_content=AsyncMock(return_value=mock_doc2_details)
+                    )
+                )
+            ),
+        ]
+
+        result = asyncio.run(self.mendeley_imp.list_collection_items("folder_id"))
+
+        expected_items = [
+            ItemResult(
+                item_id="doc1",
+                item_name="Doc Title 1",
+                item_type=ItemType.DOCUMENT,
+                item_path=[],
+                csl={},
+            ),
+            ItemResult(
+                item_id="doc2",
+                item_name="Doc Title 2",
+                item_type=ItemType.DOCUMENT,
+                item_path=[],
+                csl={},
+            ),
+        ]
+
+        expected_result = ItemSampleResult(items=expected_items, total_count=2)
+
+        self.assertEqual(result, expected_result)
```

Comment:
Will there be any order issues?

Reply:
Great point, solved that with my latest commit